src/core/packages/http/router-server-internal/src/router.ts

```diff
@@ -211,8 +212,11 @@ export class Router<Context extends RequestHandlerContextBase = RequestHandlerCo
     handler: InternalRouteHandler;
   }) {
     const hapiResponseAdapter = new HapiResponseAdapter(responseToolkit);
+    let apmSpan = (request.app as KibanaRequestState).span;
     try {
+      apmSpan = apm.startSpan('route handler');
```

Comment:
I'm not sure I follow these lines. Are we extracting the span from `request.app` and then immediately overriding it with a new span?

Do we maybe want something like:

```typescript
let apmSpan: Span | undefined;
try {
  apmSpan = apm.startSpan('route handler')?
```

Reply:
Or maybe the expect behaviour is that if `.startSpan` call fails we end the request.app span?

Reply:
Good catch, I removed the request.app.span reference.

> Or maybe the expect behaviour is that if .startSpan call fails we end the request.app span?

I am not sure about this comment. We start the span before calling the handler, and we want to stop it right after the handler's work is done. In case of an error in the handler, we ensure closing the span in the catch, and if the span is already closed, nothing happens (as mentioned [here](https://www.elastic.co/docs/reference/apm/agents/nodejs/span-api#span-end)).

Is something missing?